apps/nextjs-x-chain/src/app/components/CCTPTransfer.tsx

```diff
@@ -192,14 +190,22 @@ export function CCTPTransfer({
     transfer()
       .then((response) => {
         setTransferResponse(response);
-        setTransactionInProgress(false);
         setTransactionCompleted(true);
+        setBalanceUpdatePending(true);
+        // Use delayed refetch for cross-chain transfers - origin updates immediately, Aptos balance updates after delay
+        refetchBalancesWithDelay(8000); // 8 second delay for cross-chain processing
+
+        // Clear the pending state after the delay
+        setTimeout(() => {
+          setBalanceUpdatePending(false);
+        }, 8500);
```

Comment:
This is just an example right? Cause if this is real code we probably shouldn't do this :)

Reply:
yes, only the demo dapp. It is because we use Indexer to query for the account usdc balance, trying to see if we can use a view function for that